### ISSUE TITLE
support for git wire protocol version 2

### ISSUE THREAD

**Issue by michallula** [go-git/go-git#264]
Does go-git supports git wire protocol version 2? 

I try to use go-git with the recent git client (after 2.26) which uses git wire protocol version 2 by default, but it seems that go-git supports only version 1. Am I right?

Do you plan to add support for version 2 as well?

**Comment by simongottschlag** [comment]
The FluxCD project had to add git2go support to be able to use it: https://github.com/libgit2/git2go

**Comment by blmayer** [comment]
I think this is a good issue to work on for the next big release. What's your opinion on this @pjbgf ?

**Comment by pjbgf** [comment]
@blmayer I completely agree. If you are keen on implementing it, I will be more than happy to review the PRs.

**Comment by blmayer** [comment]
Cool, I'll try.

**Comment by blmayer** [comment]
Thanks!

**Comment by pjbgf** [comment]
I'm picking this one up.

**Comment by pjbgf** [comment]
An initial draft of the changes can be seen here: https://github.com/go-git/go-git/tree/transport-v2-capabilities.

**Comment by pjbgf** [comment]
Removed from the v6 milestone as it is not blocking the release.

**Comment by blmayer** [comment]
hello again, I worked a bit on this, please check #2188.

Thanks.
